File: deprecated/processor-pegdown/src/test/java/org/bsc/markdown/pegdown/Issue226Test.java
package org.bsc.markdown.pegdown;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.util.Arrays;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class Issue226Test extends PegdownParse {


    @Override
    protected char[] loadResource() throws IOException {
        return super.loadResource("issue226.md");
    }

    @Test
    public void parse() throws Exception {
        final String [] lines = serializeToString( createPage("home", "./home.md") ).split( "\n" );

        Arrays.asList(lines).forEach( l -> System.out.printf( "-%s\n",l) );

        int i=2;
        assertEquals( "h2.Properties", lines[i++]);
        i++;
        assertEquals( "||name ||description ||type ||default values ||since ||", lines[i++]);
        //assertEquals( "|packagePatterns |regular expression |[String\\[\\]|https://checkstyle.sourceforge.io/property_types.html#String.5B.5D] |{{[]}}|1.2.0 |", lines[i++]);
        //assertEquals( "|annotationNames |full qualified class name or simple class name of an annotation |[String\\[\\]|https://checkstyle.sourceforge.io/property_types.html#String.5B.5D]|{{[]}} |1.0.0 |", lines[i++]);

    }
}

File: markdown-processor-commonmark/src/test/kotlin/org/bsc/makdown/commonmark/Issue226Test.kt
package org.bsc.makdown.commonmark;

import org.bsc.confluence.model.Site
import org.bsc.markdown.MarkdownVisitorHelper
import org.junit.Test
import org.junit.jupiter.api.Assertions.assertEquals
import java.net.URI
import java.nio.file.Paths
import java.util.regex.Pattern

class Issue226Test {

    var site = Site().apply {
        basedir = Paths.get(System.getProperty("user.dir"))

        val home = Site.Home()
        home.uri = URI("./page.md")
        home.name = "page"

        setHome(home)

    }

    private fun parseResource():String? = parseResource(this.javaClass, "issue226", this.site)

    @Test
    fun `escape markdown text`() {
        val input = """String\[x]"""

        val leftS = Regex( """^(\\)(\[)""" )
        val rightS = Regex("""(^\\])|]""")
        val result = input.replace( leftS,"\\\\$1" ).replace(rightS, "\\\\]")

        assertEquals( """String\[x\]""", result  )

        assertEquals( """String\[x\]""", MarkdownVisitorHelper.escapeMarkdownText( """String\[x]""" ) )
        assertEquals( """String\[\]""", MarkdownVisitorHelper.escapeMarkdownText( "String[]" ) )

    }
    @Test
    fun parse() {
        val content = parseResource()

        assertEquals("""
h2. Properties
||name||description||type||default values||since||
|packagePatterns|regular expression|[String\[\]|https://checkstyle.sourceforge.io/property_types.html#String.5B.5D]|{{[]}}|1.2.0|
|annotationNames|full qualified class name or simple class name of an annotation|[String\[\]|https://checkstyle.sourceforge.io/property_types.html#String.5B.5D]|{{[]}}|1.0.0|
        """.trimIndent(), content)
    }

}
